src/storage/mod.rs

```diff
@@ -255,6 +255,152 @@ where
         }
     }
 
+    /// Tries to retrieve `e` from the storage and passes a mutable reference to `and_then`.
```

Comment:
I was really confused first by both `e` and `and_then` when I first read this. Mostly because I didn't read the function signature. If were going to use them in documentation I would rename `e` to `entity`. (`and_then` was confusing because it's name used for `Option`s `flat_map`/`bind`)

Reply:
This is probably not that huge of a deal, but somehow I was confused by it.